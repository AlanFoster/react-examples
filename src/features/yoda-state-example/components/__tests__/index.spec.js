/* eslint-env browser */
import React from 'react';
import renderer from 'react-test-renderer';
import TranslationForm from '../';

it('renders the translation form', () => {
  const onTranslation = function() {
    /* noop */
  };

  const onChange = function() {
    /* noop */
  };

  const component = renderer.create(
    <TranslationForm
      onTranslate={onTranslation}
      onChange={onChange}
      sentence={'sentence text'}
      translation={'translation text'}
      isTranslating={true}
    />
  );

  expect(component).toMatchSnapshot();
});
