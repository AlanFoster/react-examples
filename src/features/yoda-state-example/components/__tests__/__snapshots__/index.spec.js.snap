// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the translation form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Translation text
      </label>
      <input
        className="form-control"
        disabled={true}
        id={undefined}
        onChange={[Function]}
        type="text"
        value="sentence text"
      />
    </div>
    <button
      className="btn btn-default"
      disabled={true}
      type="submit"
    >
      Translate!
    </button>
  </form>
  <div>
    translation text
  </div>
</div>
`;
